subprojects/core/src/main/java/org/gradle/execution/plan/ToPlannedTaskConverter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution.plan;
+
+import org.gradle.api.internal.project.taskfactory.TaskIdentity;
+import org.gradle.initialization.DefaultPlannedTask;
+import org.gradle.internal.taskgraph.CalculateTaskGraphBuildOperationType;
+import org.gradle.internal.taskgraph.NodeIdentity;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ToPlannedTaskConverter implements ToPlannedNodeConverter {
+
+    @Override
+    public Class<? extends Node> getSupportedNodeType() {
+        return TaskNode.class;
+    }
+
+    @Override
+    public CalculateTaskGraphBuildOperationType.TaskIdentity getNodeIdentity(Node node) {
+        TaskNode taskNode = (TaskNode) node;
+        TaskIdentity<?> delegate = taskNode.getTask().getTaskIdentity();
+        return new CalculateTaskGraphBuildOperationType.TaskIdentity() {
```

Comment:
🤔 I think it would be good to extract these to actual inner classes instead of using anonymous classes. I think the code would be easier to read. WDYT?